superset/assets/visualizations/Legend.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './Legend.css';
+
+const propTypes = {
+  categories: PropTypes.object,
+  toggleCategory: PropTypes.func,
+  position: PropTypes.oneOf(['tl', 'tr', 'bl', 'br']),
+};
+
+const defaultProps = {
+  categories: {},
+  toggleCategory: () => {},
+  position: 'tr',
+};
+
+export default class Legend extends React.PureComponent {
+  render() {
+    if (Object.keys(this.props.categories).length === 0) {
+      return null;
+    }
+
+    const categories = Object.entries(this.props.categories).map(([k, v]) => {
+      const style = { color: 'rgba(' + v.color.join(', ') + ')' };
+      const icon = v.enabled ? '\u25CF' : '\u25CB';
+      return (
+        <li>
+          <a onClick={() => this.props.toggleCategory(k)}>
```

Comment:
It's not needed, but it could be nice to select only current on dblclick, and would match the nvd3 behavior.